lib/_inspect.js

```diff
@@ -170,7 +170,19 @@ class NodeInspector {
     this.domainNames.forEach((domain) => {
       this[domain] = createAgentProxy(domain, this.client);
     });
+    let startBreakReached = false;
```

Comment:
This will only work on the initial start but won't catch subsequent starts, e.g. from crashes followed by the `run` command. Is that on purpose?

One structural note: This is fairly low-level wiring code. Putting logic here seems out of place. Could you see if it's possible to put it into the handler of the pause event? https://github.com/nodejs/node-inspect/blob/4f4e7524460b42c5660aa2094adf99ce248db622/lib/internal/inspect_repl.js#L784-L806

Reply:
@jkrems good catch, I overlooked the fact that we might want to re-run it before exiting.

RE structure: good point, I don't think putting logic there is particularly clean myself. At the same time, I'm not sure putting this logic inside of the REPL makes perfect sense either... there's also no simple way to add another 'paused' listener and cancel event propagation to the REPL (which I think would make the most sense). In fact, I could come up with a way to do that, but if you want to keep things simple, I'll just put it inside this REPL listener. Your call.

Reply:
@jkrems any update on this? Just wanted to confirm that you prefer to move the logic inside the REPL listener as opposed to other options.

Reply:
Hey, sorry for losing track of this. The name "repl" may be a bit misleading here. The high-level structure here is:

* `lib/_inspect.js`: CLI parsing, process management, and general wiring things together. It doesn't really have any "business logic".
* `lib/internal`: This is where pretty much all important code belongs.
* `lib/internal/inspect_client.js`: The protocol implementation.
* `lib/internal/inspect_repl.js`: The actual command line debugger ("the inspect repl"). All business logic / user experience / UI state lives here.

So the "repl" is basically "all code" because the entire debugger is pretty much implemented in that one file. The fact that the debugger is called a "repl" is an unfortunate naming choice by past-me. :) Longer term there may be value in refactoring the debugger into multiple files but right now it's all in `inspect_repl.js`.

Reply:
Fair enough